### ISSUE TITLE
iOS Deliver - Submit app for review fails if app has an in-app event draft

### ISSUE THREAD

**Issue by pmanuelli** [fastlane/fastlane#19603]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

The deliver method fails when I try to submit for review an iOS version when I have an In-App Event draft.

I did it manually on the AppStore and when I select the option _Add for Review_ it displays a new Submission Confirmation screen (new for me at least) to select which elements I want to send for review. It displays the new version of the app and the in-app event draft.

I only want to submit the new app version, not the in-app event draft.

##### Command executed

```
+----------------------------------------------------+-------------------------+
|                           deliver 2.196.0 Summary                            |
+----------------------------------------------------+-------------------------+
| app_version                                        | 3.140.1                 |
| skip_binary_upload                                 | true                    |
| skip_screenshots                                   | true                    |
| skip_metadata                                      | true                    |
| run_precheck_before_submit                         | false                   |
| precheck_include_in_app_purchases                  | false                   |
| force                                              | true                    |
| app_identifier                                     | xxxxxxxxxxxxx           |
| build_number                                       | 3.140.1.1               |
| submit_for_review                                  | true                    |
| submission_information.add_id_info_serves_ads      | true                    |
| submission_information.add_id_info_tracks_action   | true                    |
| submission_information.add_id_info_tracks_install  | true                    |
| submission_information.add_id_info_uses_idfa       | true                    |
| submission_information.add_id_info_limits_tracking | true                    |
| api_key                                            | xxxxxxxxxxxxx           |
| screenshots_path                                   | ./fastlane/screenshots  |
| metadata_path                                      | ./fastlane/metadata     |
| username                                           | xxxxxxxxxxxxx           |
| platform                                           | ios                     |
| edit_live                                          | false                   |
| use_live_version                                   | false                   |
| skip_app_version_update                            | false                   |
| overwrite_screenshots                              | false                   |
| sync_screenshots                                   | false                   |
| reject_if_possible                                 | false                   |
| phased_release                                     | false                   |
| reset_ratings                                      | false                   |
| team_id                                            | 17233                   |
| dev_portal_team_id                                 | xxxxxxxxxxxxx           |
| precheck_default_rule_level                        | warn                    |
| ignore_language_directory_validation               | false                   |
+----------------------------------------------------+-------------------------+
```
The error is 

```
Spaceship::UnexpectedResponse: [!] Unable to submit reviewSubmissions. - Unable to submit reviewSubmissions because it has relationship to appEvents with id 6442675128.
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> 

bundler: failed to load command: fastlane (/Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/bin/fastlane)

Spaceship::UnexpectedResponse: [!] Unable to submit reviewSubmissions. - Unable to submit reviewSubmissions because it has relationship to appEvents with id 6442675128.

  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/spaceship/lib/spaceship/connect_api/api_client.rb:199:in `handle_response'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/spaceship/lib/spaceship/connect_api/api_client.rb:124:in `post'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/spaceship/lib/spaceship/connect_api/tunes/tunes.rb:983:in `post_app_store_version_submission'
  /usr/local/Cellar/ruby@2.6/2.6.6_1/lib/ruby/2.6.0/forwardable.rb:230:in `post_app_store_version_submission'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/spaceship/lib/spaceship/connect_api/models/app_store_version.rb:178:in `create_app_store_version_submission'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/deliver/lib/deliver/submit_for_review.rb:26:in `submit!'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/deliver/lib/deliver/runner.rb:206:in `submit_for_review'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/deliver/lib/deliver/runner.rb:65:in `run'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/actions/upload_to_app_store.rb:22:in `run'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
  lanes/stages/submit_for_review.rb:18:in `block in parsing_binding'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:109:in `block (2 levels) in run'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/gems/fastlane-2.196.0/bin/fastlane:23:in `<top (required)>'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/bin/fastlane:23:in `load'
  /Users/macos/gitlab-runner/builds/etermax/preguntados/frontend/ios/.bundle/ruby/2.6.0/bin/fastlane:23:in `<top (required)>' </pre>
</details>

### Environment

<details>
  <pre> <details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 11.4                                             |
| Ruby                        | 2.6.3                                            |
| Bundler?                    | true                                             |
| Git                         | git version 2.30.1 (Apple Git-130)               |
| Installation Source         | ~/.rbenv/versions/2.6.3/bin/fastlane             |
| Host                        | macOS 11.4 (20F71)                               |
| Ruby Lib Dir                | ~/.rbenv/versions/2.6.3/lib                      |
| OpenSSL Version             | OpenSSL 1.1.1i  8 Dec 2020                       |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode_13.1.app/Contents/Developer/ |
| Xcode Version               | 13.1                                             |
| Swift Version               | 5.5.1                                            |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.197.0 | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.3.0        |
| bundler                         | 1.17.2       |
| etc                             | 1.0.1        |
| forwardable                     | 1.2.0        |
| rake                            | 13.0.6       |
| rexml                           | 3.2.5        |
| CFPropertyList                  | 3.0.4        |
| ZenTest                         | 4.12.0       |
| RubyInline                      | 3.12.5       |
| concurrent-ruby                 | 1.1.9        |
| i18n                            | 1.8.10       |
| minitest                        | 5.14.4       |
| tzinfo                          | 2.0.4        |
| zeitwerk                        | 2.4.2        |
| activesupport                   | 6.1.4.1      |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| httpclient                      | 2.8.3        |
| json                            | 2.6.1        |
| algoliasearch                   | 1.27.5       |
| artifactory                     | 3.0.15       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.2.0        |
| aws-partitions                  | 1.525.0      |
| aws-sigv4                       | 1.4.0        |
| jmespath                        | 1.4.0        |
| aws-sdk-core                    | 3.122.0      |
| aws-sdk-kms                     | 1.51.0       |
| aws-sdk-s3                      | 1.105.0      |
| babosa                          | 1.0.4        |
| claide                          | 1.0.3        |
| fuzzy_match                     | 2.0.4        |
| nap                             | 1.1.0        |
| netrc                           | 0.11.0       |
| ffi                             | 1.15.4       |
| ethon                           | 0.14.0       |
| typhoeus                        | 1.4.0        |
| cocoapods-core                  | 1.11.2       |
| cocoapods-deintegrate           | 1.0.5        |
| cocoapods-downloader            | 1.5.1        |
| cocoapods-plugins               | 1.0.0        |
| cocoapods-search                | 1.0.1        |
| cocoapods-trunk                 | 1.6.0        |
| cocoapods-try                   | 1.2.0        |
| colored2                        | 3.1.2        |
| escape                          | 0.0.4        |
| fourflusher                     | 2.3.1        |
| gh_inspector                    | 1.1.3        |
| molinillo                       | 0.8.0        |
| ruby-macho                      | 2.5.1        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.21.0       |
| cocoapods                       | 1.11.2       |
| dotenv                          | 2.7.6        |
| osx_keychain                    | 1.0.2        |
| cocoapods-keys                  | 2.2.1        |
| colored                         | 1.2          |
| colorize                        | 0.8.1        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.4        |
| unf_ext                         | 0.0.8        |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.88.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| multipart-post                  | 2.0.0        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.8.0        |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.5        |
| jwt                             | 2.3.0        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.1        |
| signet                          | 0.16.0       |
| googleauth                      | 1.1.0        |
| mini_mime                       | 1.1.2        |
| trailblazer-option              | 0.1.1        |
| uber                            | 0.1.0        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.4.1        |
| google-apis-androidpublisher_v3 | 0.13.0       |
| google-apis-playcustomapp_v1    | 0.6.0        |
| google-apis-iamcredentials_v1   | 0.8.0        |
| google-apis-storage_v1          | 0.9.0        |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-cloud-storage            | 1.34.1       |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.1.1        |
| plist                           | 3.6.0        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| simctl                          | 1.6.8        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
| rchardet                        | 1.8.0        |
| git                             | 1.9.1        |
| xml-simple                      | 1.1.5        |
| fastlane-plugin-bugsnag         | 2.2.0        |
| http-form_data                  | 2.3.0        |
| ffi-compiler                    | 1.0.1        |
| llhttp-ffi                      | 0.4.0        |
| http                            | 5.0.4        |
| http-accept                     | 1.7.0        |
| mime-types-data                 | 3.2021.0901  |
| mime-types                      | 3.3.1        |
| rest-client                     | 2.1.0        |
| hashie                          | 3.6.0        |
| mini_portile2                   | 2.6.1        |
| racc                            | 1.6.0        |
| nokogiri                        | 1.12.5       |
| unity_test_results              | 0.1.1        |
| fastlane-plugin-etermax_builds  | 4.14.17      |
| multi_xml                       | 0.6.0        |
| httparty                        | 0.18.1       |
| gitlab                          | 4.7.0        |
</details>

*generated on:* **2021-11-12**
</details> </pre>
</details>

**Comment by Onno113** [comment]
We just had the case where submit for review failed because the review for the in-app event was not done after four days waiting for review.

**Comment by pmanuelli** [comment]
@Onno113 I'm glad that you manage to solve your issue.

I'm still struggling with this. In my case, I do not want to send the in-app event draft to review. Just the new app version.

What parameters are you passing to `deliver` in order to send both app and in-app event to review? Maybe that will give me some clue about how to solve my problem.

I think that I'm just missing something here.

Thank you!

**Comment by Onno113** [comment]
@pmanuelli we did all the in-app events stuff manually, not with fastlane. 

As I wrote we send both app and in-app event together in review, but then the review took very long, more than a week. We cancelled it again, because it took too long for us and Apple didn't give us a hint why it took so long.

Currently we are back to submitting the app with fastlane. When the app is approved, it gets released and then our responsible person from marketing submits the in-app event. I am not 100% sure, but I think we don't create in-app drafts until the app is released. So we try to keep both things separate as much as we can for now.

**Comment by pmanuelli** [comment]
Oh, I understand now @Onno113. Thank you for the clarification.

We do the same thing. Send the version via fastlane and only after is approved create and submit the in-app event. But sometimes happens that marketing creates the event before and our pipeline breaks when sending the app to review.

I really don't know if this issue is already fixed on fastlane. I read the `deliver` [documentation](https://docs.fastlane.tools/actions/deliver/) but in-app events it's not mentioned. I imagine that this issue must be affecting other teams too.

**Comment by ChristopherDrum** [comment]
Just had the same issue, except for "custom product pages."
Found this notice on Apple's site. This "updated submission experience" appears to have been given to everyone here at the end of the year. It appears to alter/break the fastlane workflow under certain conditions (i.e. there can now be more than one item pending submission)
https://developer.apple.com/app-store-connect/submission-update/

**Comment by StarLard** [comment]
Our team is also encountering this when having an in-app event in review. Any movement on this?

**Comment by pmanuelli** [comment]
Hello @StarLard. Did you try the latest version of Fastlane? I think that the issue is solved. 

The Fastlane API does not allow sending the in-app events along with the version yet. We are still doing it manually.

**Comment by StarLard** [comment]
Thanks for the response. I see there is a new version available; I will try updating to that and get back if it fixes the issue.

**Comment by howadev** [comment]
I tried to update to latest Fastlane (2.204.3) but still got the error if there is "custom product pages" pending. Had to remove it to submit the app for review.

**Comment by StarLard** [comment]
@pmanuelli We updated our project to use Fastlane `2.204.3` and we've just run into the same issue, so it seems the update didn't fix the issue. I'll post any other updates as I have them. Let me know if I can help in any other way.

**Comment by StarLard** [comment]
Can we get this issue reopened? It's still occurring after updating Fastlane.
